WordPress/Classes/ViewRelated/Gutenberg/Processors/GutenbergCoverUploadProcessor.swift

```diff
@@ -0,0 +1,63 @@
+import Foundation
+import Aztec
+
+class GutenbergCoverUploadProcessor: Processor {
+    public typealias InnerBlockProcessor = (String) -> String?
+
+    let mediaUploadID: Int32
+    let remoteURLString: String
+    let serverMediaID: Int
+
+    init(mediaUploadID: Int32, serverMediaID: Int, remoteURLString: String) {
+        self.mediaUploadID = mediaUploadID
+        self.serverMediaID = serverMediaID
+        self.remoteURLString = remoteURLString
+    }
+
+    lazy var coverBlockProcessor = GutenbergBlockProcessor(for: "wp:cover", replacer: { coverBlock in
+        guard let mediaID = coverBlock.attributes["id"] as? Int,
+            mediaID == self.mediaUploadID else {
+                return nil
+        }
+        var block = "<!-- wp:cover "
+
+        var attributes = coverBlock.attributes
+        attributes["id"] = self.serverMediaID
+        attributes["url"] = self.remoteURLString
+
+        if let jsonData = try? JSONSerialization.data(withJSONObject: attributes, options: [.sortedKeys]),
+            let jsonString = String(data: jsonData, encoding: .utf8) {
+            block += jsonString
+        }
+
+        block += " -->"
+        block += self.htmlUploadProcessor.process(coverBlock.content)
+        block += "<!-- /wp:cover -->"
+        return block
+    })
+
+    lazy var htmlUploadProcessor = HTMLProcessor(for: "div", replacer: { (div) in
+
+        guard let styleAttributeValue = div.attributes["style"]?.value,
+            case let .string(styleAttribute) = styleAttributeValue,
+            styleAttribute.hasPrefix("background-image:url(file:///") else {
```

Comment:
I wonder if the assumption will hold that this background-image style will be _first_ in the list of inline styles. E.g. the following test will fail currently:

<details>
<summary>Test with background-image not first</summary>

```swift
    let postContent2 = """
       <!-- wp:cover {"url":"file:///usr/tmp/-1175513456.jpg","id":-1175513456} -->
       <div class="wp-block-cover has-background-dim" style="color:black;background-image:url(file:///usr/tmp/-1175513456.jpg)"><div class="wp-block-cover__inner-container">
       <!-- wp:paragraph {"align":"center","placeholder":"Write title…"} -->
       <p class="has-text-align-center"></p>
       <!-- /wp:paragraph -->
       </div></div>
       <!-- /wp:cover -->
       """

       let postResultContent2 = """
       <!-- wp:cover {"id":456,"url":"http:\\/\\/www.wordpress.com\\/logo.jpg"} -->
       <div class="wp-block-cover has-background-dim" style="color:black;background-image:url(http://www.wordpress.com/logo.jpg)"><div class="wp-block-cover__inner-container">
       <!-- wp:paragraph {"align":"center","placeholder":"Write title…"} -->
       <p class="has-text-align-center"></p>
       <!-- /wp:paragraph -->
       </div></div>
       <!-- /wp:cover -->
       """

       func testCoverBlockProcessor2() {
           let gutenbergMediaUploadID = Int32(-1175513456)
           let mediaID = 456
           let remoteURLStr = "http://www.wordpress.com/logo.jpg"

           let gutenbergCoverPostUploadProcessor = GutenbergCoverUploadProcessor(mediaUploadID: gutenbergMediaUploadID, serverMediaID: mediaID, remoteURLString: remoteURLStr)
           let resultContent = gutenbergCoverPostUploadProcessor.process(postContent2)

           XCTAssertEqual(resultContent, postResultContent2, "Post content should be updated correctly")
       }
```

Similarly (and also with a change possibly needed for the [WordPress-Android processor](https://github.com/wordpress-mobile/WordPress-Android/blob/29d19b42b325dfdaa8445eebc4b7104968abd633/WordPress/src/main/java/org/wordpress/android/ui/posts/mediauploadcompletionprocessors/CoverBlockProcessor.java#L24)), this would fail when there is a space after the colon.

Reply:
That’s a good point a Regex here would probably be the safer option.

Reply:
I agree that a regex will more failsafe here.